esp-hal/src/reset.rs

```diff
@@ -1,32 +1,19 @@
 //! # Hardware and Software Reset
 //!
 //! ## Overview
-//! The Hardware and Software Reset module provides functions for performing
-//! hardware and software resets on ESP chips. It also includes functions for
+//! Espressif chips provide four types of reset that occur at different levels,
+//! namely CPU Reset, Core Reset, System Reset, and Chip Reset. All reset types
```

Comment:
Just a suggestion, feel free to decline: maybe let's put those mods to ``?
(Like `CPU Reset`)